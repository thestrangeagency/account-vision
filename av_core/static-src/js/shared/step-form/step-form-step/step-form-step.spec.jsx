import React from 'react';
import { mount } from 'enzyme';

import StepFormStep from './step-form-step';

describe('StepFormStep', () => {
  let wrapper;
  let callback;

  beforeEach(() => {
    callback = jest.fn();

    const inputs = [
      {
        type: 'TEXT',
        inputProps: {
          label: 'First Name',
          name: 'first_name',
          required: false,
        },
      },
    ];

    wrapper = mount((
      <StepFormStep
        hasNextButton
        id="foo"
        inputs={inputs}
        onClickNext={callback}
        onSetValues={() => {}}
      />
    ));
  });

  test('it renders', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('it can be focused', () => {
    const input = wrapper.find('input').at(0).instance();
    const spy = jest.spyOn(input, 'focus');
    wrapper.setProps({ isFocused: true });
    expect(spy).toHaveBeenCalled();
  });

  test('it calls callback when user presses enter key', () => {
    const spy = jest.spyOn(wrapper.instance(), 'onAdvance');
    wrapper.find('.step-form-step').simulate('keyup', { which: 13 });
    expect(spy).toHaveBeenCalled();
  });

  test('it checks for validity', () => {
    wrapper.setProps({
      inputProps: [
        {
          type: 'TEXT',
          inputProps: {
            label: 'First Name',
            name: 'first_name',
            required: true,
          },
        },
      ],
    });

    const nextButton = wrapper.find('button');
    nextButton.simulate('click');
    expect(wrapper).toMatchSnapshot();

    const input = wrapper.find('input').at(0).instance();
    input.value = 'foo';
    nextButton.simulate('click');
    expect(wrapper).toMatchSnapshot();
  });
});
