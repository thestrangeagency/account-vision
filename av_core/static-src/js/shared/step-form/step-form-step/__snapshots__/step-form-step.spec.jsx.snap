// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StepFormStep it checks for validity 1`] = `
<StepFormStep
  hasBackButton={false}
  hasNextButton={true}
  id="foo"
  inputProps={
    Array [
      Object {
        "inputProps": Object {
          "label": "First Name",
          "name": "first_name",
          "required": true,
        },
        "type": "TEXT",
      },
    ]
  }
  inputs={
    Array [
      Object {
        "inputProps": Object {
          "label": "First Name",
          "name": "first_name",
          "required": false,
        },
        "type": "TEXT",
      },
    ]
  }
  isDisabled={false}
  isFocused={false}
  nextText="Next"
  onClickBack={null}
  onClickNext={[Function]}
  onSetValues={[Function]}
  prevText="← Back"
>
  <div
    aria-selected={false}
    className="step-form-step row step-form-step--hidden"
    id="foo"
    onKeyUp={[Function]}
    role="option"
    tabIndex="-1"
  >
    <div
      className="col"
    >
      <div
        className="form-group"
        key="0"
      >
        <label
          htmlFor="input_first_name"
        >
          First Name
        </label>
        <input
          className="form-control"
          defaultValue={undefined}
          id="input_first_name"
          name="first_name"
          pattern={undefined}
          placeholder={undefined}
          required={false}
          type="text"
        />
        <div
          className="invalid-feedback"
        >
          This field is required.
        </div>
      </div>
      <div
        className="row justify-content-end"
      >
        <div
          className="col col-6"
        >
          <button
            className="btn btn-primary w-100"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </div>
</StepFormStep>
`;

exports[`StepFormStep it checks for validity 2`] = `
<StepFormStep
  hasBackButton={false}
  hasNextButton={true}
  id="foo"
  inputProps={
    Array [
      Object {
        "inputProps": Object {
          "label": "First Name",
          "name": "first_name",
          "required": true,
        },
        "type": "TEXT",
      },
    ]
  }
  inputs={
    Array [
      Object {
        "inputProps": Object {
          "label": "First Name",
          "name": "first_name",
          "required": false,
        },
        "type": "TEXT",
      },
    ]
  }
  isDisabled={false}
  isFocused={false}
  nextText="Next"
  onClickBack={null}
  onClickNext={[Function]}
  onSetValues={[Function]}
  prevText="← Back"
>
  <div
    aria-selected={false}
    className="step-form-step row step-form-step--hidden"
    id="foo"
    onKeyUp={[Function]}
    role="option"
    tabIndex="-1"
  >
    <div
      className="col"
    >
      <div
        className="form-group"
        key="0"
      >
        <label
          htmlFor="input_first_name"
        >
          First Name
        </label>
        <input
          className="form-control"
          defaultValue={undefined}
          id="input_first_name"
          name="first_name"
          pattern={undefined}
          placeholder={undefined}
          required={false}
          type="text"
        />
        <div
          className="invalid-feedback"
        >
          This field is required.
        </div>
      </div>
      <div
        className="row justify-content-end"
      >
        <div
          className="col col-6"
        >
          <button
            className="btn btn-primary w-100"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </div>
</StepFormStep>
`;

exports[`StepFormStep it renders 1`] = `
<StepFormStep
  hasBackButton={false}
  hasNextButton={true}
  id="foo"
  inputs={
    Array [
      Object {
        "inputProps": Object {
          "label": "First Name",
          "name": "first_name",
          "required": false,
        },
        "type": "TEXT",
      },
    ]
  }
  isDisabled={false}
  isFocused={false}
  nextText="Next"
  onClickBack={null}
  onClickNext={[Function]}
  onSetValues={[Function]}
  prevText="← Back"
>
  <div
    aria-selected={false}
    className="step-form-step row step-form-step--hidden"
    id="foo"
    onKeyUp={[Function]}
    role="option"
    tabIndex="-1"
  >
    <div
      className="col"
    >
      <div
        className="form-group"
        key="0"
      >
        <label
          htmlFor="input_first_name"
        >
          First Name
        </label>
        <input
          className="form-control"
          defaultValue={undefined}
          id="input_first_name"
          name="first_name"
          pattern={undefined}
          placeholder={undefined}
          required={false}
          type="text"
        />
        <div
          className="invalid-feedback"
        >
          This field is required.
        </div>
      </div>
      <div
        className="row justify-content-end"
      >
        <div
          className="col col-6"
        >
          <button
            className="btn btn-primary w-100"
            disabled={false}
            onClick={[Function]}
            type="button"
          >
            Next
          </button>
        </div>
      </div>
    </div>
  </div>
</StepFormStep>
`;
